core/server/inventory.go

```diff
@@ -446,3 +436,26 @@ func parseInventoryFromUnstructured(obj *unstructured.Unstructured) ([]*unstruct
 
 	return objects, nil
 }
+
+func secretNameFromHelmRelease(helmRelease *helmv2.HelmRelease) *client.ObjectKey {
```

Comment:
The logic below continues to be problematic, as one of the things `v2beta2` addressed is field mutations which change the target installation of a release.

When someone changes their `.spec.targetNamespace`, `.spec.storageNamespace` or `.spec.releaseName`. The controller will now first uninstall this release before installing the new one. Given this, you _have_ to take the `.status.storageNamespace` and data from `.status.history` into account to get an accurate view of the active release. As it may take some time for the controller to perform a successful uninstall, which means the data taken from `.spec` may not be an accurate representation of the active state, it's _desired_ state.